ArduCopter/control_guided.cpp

```diff
@@ -212,6 +213,30 @@ void Copter::guided_set_angle(const Quaternion &q, float climb_rate_cms)
     guided_angle_state.update_time_ms = millis();
 }
 
+// guided_reached_destination_notify - notify if we've reached the destination
+// on guided mode - if we're taking off we consider altitude
+void Copter::guided_reached_destination_notify(bool check_altitude)
+{
+    bool reached_alt = true;
+
+    if (guided_prev_mode != guided_mode)
+        guided_reached_notified = false;
```

Comment:
```prev_mode``` is indeed used to detect transitions between modes - currently mostly interested on takeoff and WP. What would be the better way to detect that?

Thanks...